pkg/agent/agent.go

```diff
@@ -1144,8 +1144,11 @@ func (i *Initializer) getNodeInterfaceFromIP(nodeIPs *utilip.DualStackIPs) (v4IP
 // getFreeOFPort returns an OpenFlow port number which is not used by any existing OVS port. Note that, the returned port
 // is not saved in OVSDB yet before the real port is created, so it might introduce an issue for the same return value
 // if it is called multiple times before OVS port creation.
-func (i *Initializer) getFreeOFPort(startPort int) (int32, error) {
+func (i *Initializer) getFreeOFPort(startPort int, reservedPorts []int32) (int32, error) {
```

Comment:
nit: `getFreeOFPort(startPort int, reservedPorts ...int32)` also makes sense in this case and can avoid updating callers which don't need to set `reservedPorts`.

Reply:
To make it simpler, should we maintain the allocated port numbers in OVSBridgeClient (so no need to have the callers pass the reservedPorts)?

Reply:
Accepted Quan's comment. We can discuss here https://github.com/antrea-io/antrea/pull/3938#discussion_r914290095 for reservedPorts cache.